lib/puppet/application/master.rb

```diff
@@ -186,10 +186,11 @@ def main
 
     if Puppet.features.root?
       begin
-        Puppet::Util.chuser
+        if Etc.getpwnam(Puppet[:user])
```

Comment:
First approach: use `getpwnam` which throws an ArgumentError if the user doesn't exist, and emit a warning if it does.